src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -172,12 +177,22 @@ public Task<uint256> GetTrxBlockIdAsync(uint256 trxid)
         }
 
         /// <inheritdoc/>
-        public Task<Block> GetBlockAsync(uint256 blockHash)
+        public async Task<Block> GetBlockAsync(uint256 blockHash)
         {
-            // TODO ACTIVATION take blocks from pending
-            // TODO make sure this method is used only by CM and all components ask the blocks from CM
+            this.logger.LogTrace("({0}:'{1}')", nameof(blockHash), blockHash);
+
+            Block block = null;
+
+            lock (this.batchLock)
+            {
+                block = this.batch.FirstOrDefault(x => x.ChainedHeader.HashBlock == blockHash)?.Block;
+            }
 
-            return this.blockRepository.GetBlockAsync(blockHash);
+            if (block == null)
+                block = await this.blockRepository.GetBlockAsync(blockHash);
+
+            this.logger.LogTrace("(-)");
```

Comment:
We should log the block to know whether it was found or not

Reply:
fixed